drivers/flash/nrf_qspi_nor.c

```diff
@@ -1041,6 +1050,100 @@ static const struct flash_driver_api qspi_nor_api = {
 #endif /* CONFIG_FLASH_JESD216_API */
 };
 
+#ifdef CONFIG_PM_DEVICE
+static int enter_dpd(const struct device *const dev)
+{
+	if (IS_ENABLED(DT_INST_PROP(0, has_dpd))) {
+		struct qspi_cmd cmd = {
+			.op_code = SPI_NOR_CMD_DPD,
+		};
+		int ret;
+
+		ret = qspi_send_cmd(dev, &cmd, false);
+		if (ret < 0) {
+			return ret;
+		}
+#if defined(T_ENTER_DPD_US)
```

Comment:
Wouldn't it be nicer to do it the same way line 1056 does?

Reply:
Do you think maybe that the ANOMALY_122 macros should be rewritten similar to this function, to keep similar style for conditionally included functionality?

Reply:
> Wouldn't it be nicer to do it the same way line 1056 does?

It would be, but it's not possible in this case. When the `"t_enter_dpd"` property is not provided, the macro call:
`DT_INST_PROP(0, t_enter_dpd)` cannot be compiled, so it needs to be cut out at the preprocessing stage. That's why `T_ENTER_DPD_US` is conditionally defined.

Reply:
Something like `if (DT_INSTP_PROP_OR(0, t_enter_dpd, FALSE))` wouldn't work?

Reply:
There is also this thing: `DT_INST_NODE_HAS_PROP`

Reply:
> Something like `if (DT_INSTP_PROP_OR(0, t_enter_dpd, FALSE))` wouldn't work?

Right, I haven't thought about using `DT_INST_PROP_OR` which would actually hide this conditional compilation and make the whole thing look nicer. Let me update this PR.

Reply:
Done.